### PR TITLE
Added getter for _prefix field which allows applications that want to check if the prefix has been set or not.

### PR DESCRIPTION
**Pull request by dkhanaferov**
The use case is:
1. Some initialization code configures StatsdClient with MetricsConfig instance
2. Some other part of the application wants to publish the metrics with the now configured client

If the publisher wants to check that the prefix has been set or not, currently there is not way to do that. It can be important to be able to check if prefix was set because it can potentially change the format of the published metric name and thus cause the metrics database to generate improper graphs.

By allowing the getter to retrieve the prefix, the application can intelligently decide whether or not to publish the metric or to configure the correct prefix first.

### CODE REVIEW

**Comment by DarrellMozingo** [comment]
Hi David! Thanks for the PR.

Can I ask more about the use case? The first part makes sense, but why would the publisher need to check if the prefix is set or not? It's usually set at `MetricsConfig` setup during app startup (as you mentioned) and contains a value to delineate the environment (dev/prod/region/etc). Are you changing the prefix each time you send stats? Is it somehow getting unset for you between setting up the the `MetricsConfig` setup and going to use it?

**Comment by dkhanaferov** [comment]
Yes, that's exactly right, the configuration is set on application startup and the prefix we use is basically what you describe (dev/prod/region/etc). The use case we ran into is, we have a library (pulled in via nuget) that we use for communication with services, it's a generic rest client that contains within it, statsd metrics code so our applications don't need to worry about publishing outbound call metrics, and let this library collect counter and timer metrics for all outbound calls.

Basically, in this library we want to know  (we want to be 100% sure) that the application has already configured statsd and if it hasn't we want to not allow publishing of outbound call metrics because without the proper prefix metrics will be published with incorrect name which leads to confusion when we try to roll up and plot these metrics.

So from the libraries standpoint, there is no way, right now, to determine that the prefix had been set already. If there was a getter for this field, then we can read the prefix field and see if it's null/empty/whitespace or if it does not match some regex pattern which would indicate to us that it's following our established metrics name standard.

To answer your question, no we don't want to be able to change the prefix arbitrarily, we want to be able to read it, and make sure it's been configured, and then decide whether or not to publish metric or to ask caller to configure statsd first.

**Comment by DarrellMozingo** [comment]
Thanks for explaining, that makes more sense. It sounds like you really want to check if the `MetricsConfig` class has been setup yet, correct? The prefix being set is incidental to that since you set the prefix when the class gets setup. Can I suggest an `IsConfigured` property that checks for `_statsD == null` instead? I think that would enable this use case more generally, ie for others that don't use the prefix property.

**Comment by dkhanaferov** [comment]
Makes sense, I do like the IsConfigured property better, I will go ahead and make this change.

**Comment by DarrellMozingo** [comment]
Where you able to make that change @dkhanaferov?

**Comment by dkhanaferov** [comment]
@DarrellMozingo Wow I totally forgot about this PR. I did however just make the changes and updated this PR. Check it out.

Thanks,

File: src/Tests/MetricsTests.cs [hunk 1]
```diff
@@ -1,15 +1,46 @@
...
+		}
+
+		[Test]
+		public void IsConfigured_returns_true_when_configuration_is_not_null()
+		{
+			//Configure metrics
+			Metrics.Configure(new MetricsConfig());
+
+			//Validate
+			Assert.IsFalse(Metrics.IsConfigured());
```
Comment (DarrellMozingo) [inline src/Tests/MetricsTests.cs]: I think you meant `IsTrue`? :)  Would you also mind removing the two comments? I think your code is spot on and self explanatory.

File: src/Tests/MetricsTests.cs [hunk 2]
```diff
@@ -1,15 +1,46 @@
...
+		[Test]
+		public void throws_when_configured_with_a_null_configuration()
+		{
+			Assert.Throws<ArgumentNullException>(() => Metrics.Configure(null));
+		}
+
+		[Test]
+		public void IsConfigured_returns_false_when_configuration_is_null()
+		{
+			Assert.IsFalse(Metrics.IsConfigured());
```
Comment (DarrellMozingo) [inline src/Tests/MetricsTests.cs]: Would you mind using the `Assert.That(...)` syntax for consistency? So `Assert.That(Metrics.IsConfigured(), Is.False)` and `Assert.That(() => Metrics.Configure(null), Throws.Exception.TypeOf<ArgumentNullException>)`

**Comment by dkhanaferov** [comment]
@DarrellMozingo Just made your suggested changes.

**Comment by DarrellMozingo** [comment]
Thanks David, much appreciated! I'll push this out to NuGet tomorrow.

**Pull request merged**